bundle/config/mutator/run_as_test.go

```diff
@@ -141,6 +142,7 @@ func TestRunAsErrorForUnsupportedResources(t *testing.T) {
 		"registered_models",
 		"experiments",
 		"schemas",
+		"apps",
```

Comment:
We should not include apps in the `run_as` list.

If you use top-level `run_as` and have an app, you could assume the app will run as the specified user or SP.

This is not the case today. And if apps ever adds identity assumption in the future, it won't be compatible.

Reply:
Does this test not work well for apps?

Having it on the allow lists suggests it is OK to use `run_as`, but that doesn't match having the check in the other file to reject `run_as`.

Reply:
It appears it was possible to add resource in both allowlist and also for it to return error from run_as mutator. Added `TestRunAsNoErrorForSupportedResources` to catch this and for completness